cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/ZookeeperScaler.java

```diff
@@ -0,0 +1,351 @@
+    public void close() {
+        closeConnectionSync();
+
+        if (trustStoreFile != null) {
+            if (!trustStoreFile.delete())   {
+                log.debug("Failed to delete file {}", trustStoreFile);
+            }
+        }
+
+        if (keyStoreFile != null)   {
+            if (!keyStoreFile.delete())   {
+                log.debug("Failed to delete file {}", keyStoreFile);
+            }
+        }
+    }
+
+    /**
+     * Internal method used to create the Zookeeper Admin client and connect it to Zookeeper
+     *
+     * @return      Future indicating success or failure
+     */
+    private Future<Void> connect()    {
+        Promise<Void> connected = Promise.promise();
+
+        try {
+            long timer = vertx.setTimer(operationTimeoutMs, ignored -> {
+                if (!connected.future().isComplete()) {
+                    log.debug("Failed to connected to {} to scale Zookeeper for {} ms. The connection will be closed.", this.zookeeperConnectionString, operationTimeoutMs);
+                    closeConnectionSync();
+                    connected.tryFail(new RuntimeException("Failed to connect to Zookeeper " + this.zookeeperConnectionString + " for " + operationTimeoutMs + " ms"));
+                }
+            });
+
+            zkAdmin = zooAdminProvider.createZookeeperAdmin(this.zookeeperConnectionString, 10_000, watchedEvent -> {
+                switch (watchedEvent.getState()) {
+                    case SyncConnected:
+                        if (!connected.future().isComplete()) {
+                            log.debug("Successfully connected to {} to scale Zookeeper", zookeeperConnectionString);
+                            connected.tryComplete();
+                        }
+                        break;
+                    default:
+                        if (!connected.future().isComplete()) {
+                            log.warn("Failed to connect to {} to scale Zookeeper because of state {}", zookeeperConnectionString, watchedEvent.getState());
+                            closeConnectionSync();
+                            connected.tryFail(new RuntimeException("Failed to connect to Zookeeper " + zookeeperConnectionString + " - got state " + watchedEvent.getState()));
+                        }
+                }
+
+                vertx.cancelTimer(timer);
+            }, getClientConfig());
+        } catch (IOException e)   {
+            log.warn("Failed to connect to {} to scale Zookeeper", zookeeperConnectionString, e);
+            connected.tryFail(new RuntimeException("Failed to connect to Zookeeper " + zookeeperConnectionString, e));
+        }
+
+        return connected.future();
+    }
+
+    /**
+     * Internal method to scale Zookeeper up or down or check configuration. It will:
+     *     1) Compare the current configuration with the desired configuration
+     *     2) Update the configuration if needed
+     *
+     * @param currentServers    Current configuration
```

Comment:
Why `currentServers` and not `currentConfiguration`?

Reply:
I fix the docs rather than renaming the parameter. It is only the servers part of the configuration which is passed to the `reconfigure()` method - withotu the version etc. So I thought this better distinguishes it.